hundred_x/client.py

```diff
@@ -32,13 +32,15 @@
 class HundredXClient:
     def __init__(
         self,
+        private_key: str,
         env: Environment = Environment.TESTNET,
-        private_key: str = None,
```

Comment:
Why enforce this?

There are unauthenticated endpoints.

A better solution would be the introduction of exceptions,

And then specifically raising the error to the user,

I do agree that we should 100% have some sort of enforcement of authenticated enpoints.

What about using an attribute on the client to indicate which endpoints are authenticated and raising if the private key is none? 

Would be happy to work on this with you?

Reply:
I didn't even think of that. should be easy throw errors on functions that need private key.